packages/xsnap/src/xsnap.c

```diff
@@ -1506,6 +1533,14 @@ void fxFreeSlots(txMachine* the, void* theSlots)
 	c_free(theSlots);
 }
 
+void fx_gettimeofday(txMachine *the)
+{
+	c_timeval tv;
+	c_gettimeofday(&tv, NULL);
+	mxResult->kind = XS_NUMBER_KIND;
+	mxResult->value.number = (double)(tv.tv_sec) + ((double)(tv.tv_usec) / 1000000.0);
```

Comment:
Based on the 1e6 in the nanosecond term, I think you’re shooting for units of milliseconds and accepting some loss of precision. Node’s `process.hrtime()` returns a duple of tv_sec and tv_usec, `process.hrtime.bigint()` is better, and `performance.now()` is in units of milliseconds and drops some precision deliberately.

Looks like you need to multiply the seconds to get them into milliseconds.

```suggestion
	mxResult->value.number = (double)(tv.tv_sec*1000) + ((double)(tv.tv_usec) / 1000000.0);
```

May I recommend `fx_performance_now` since this is identical to the web?

Reply:
In an earlier draft, the C function was called `fx_performance_now` but it was exposed to the start compartment as `now` because I didn't want to figure out how to make an object with a property. @warner requested a more high-entropy name.

I also didn't bother to convert from seconds to microseconds to match Performance.now; @warner prefers seconds anyway. We agreed `gettimeofday` was a fitting name. Would you two please talk among yourselves and let me know the result?

p.s. nanosecond term? I don't see one of those.

Reply:
I yield to @kriskowal 's experience and JS engine conventions.

Reply:
@warner deferred to you, @kriskowal . The trick is that this is due for today's release, presumably at the testnet meeting in 20 minutes; I'm not available for a bit for reasons I mentioned.

@michaelfig @rowgraus be advised.

Reply:
Oh, I see. usec is µsec is microseconds, not nanoseconds, so I read this wrong. Per out of band conversation, I’ll shift both terms to milliseconds, introduce a `performance` object to carry `performance.now` and push a commit.